rescript/get_data.py

```diff
@@ -12,13 +12,132 @@
+    # For each DOI, get download URL of file
+    for DOI in DOIs:
+        query_data = requests.get(base_url + DOI).json()
+        # Updates can be made to files in a DOI, so on the advice of the devs,
+        # only return the last file uploaded with this -1  vv
+        URL = query_data['data'][0]['attributes']['media'][-1]['url']
+        URLs.add(URL)
+    return URLs
+
+
+def _unite_get_url(version, taxon_group, singletons):
+    '''Generate UNITE urls, given database version and reference target.'''
+    # Lookup DOIs for various databases, source: https://unite.ut.ee/repository.php
+    unite_dois = {
+        '9.0': {'fungi':      {False: '10.15156/BIO/2938079', True: '10.15156/BIO/2938080'},
+                'eukaryotes': {False: '10.15156/BIO/2938081', True: '10.15156/BIO/2938082'}},
+        # Old version 9.0 is not listed here
+        '8.3': {'fungi':      {False: '10.15156/BIO/1264708', True: '10.15156/BIO/1264763'},
+                'eukaryotes': {False: '10.15156/BIO/1264819', True: '10.15156/BIO/1264861'}},
+        '8.2': {'fungi':      {False: '10.15156/BIO/786385', True: '10.15156/BIO/786387'},
+                'eukaryotes': {False: '10.15156/BIO/786386', True: '10.15156/BIO/786388'}},
+        '8.0': {'fungi':      {False: '', True: '10.15156/BIO/786349'},
+                'eukaryotes': {False: '', True: ''}},  # All other 8.0 are in zip files
+    }
+    # There's got to be a better way! See https://stackoverflow.com/questions/25833613/safe-method-to-get-value-of-nested-dictionary
+    try:
+        # Check if we have the DOI requested
+        target_doi = unite_dois[version][taxon_group][singletons]
+    except KeyError as ke:
+        print('Unknown DOI for this value: ' + str(ke))
+        raise
+    return _unite_dois_to_urls(target_doi).pop()
+
+_unite_get_url(version='9.0', taxon_group='fungi', singletons=False)
+
+# with tempfile.TemporaryDirectory() as tmp_dir:
+tmp_dir = tempfile.mkdtemp()
+
+def _unite_download_targz(url, download_path):
+    print('Downloading ' + url)
+        
+    response = requests.get(url, stream=True)
+    if response.status_code != 200:
+        raise ValueError("Failed to download the file from " + url)
+    
+    tar_file_path = os.path.join(download_path, 'unitefile.tar.gz')
+    with open(tar_file_path, 'wb') as f:
+        f.write(response.content)
+    
+    # Extract only the 'developer' subdirectory
+    with tarfile.open(tar_file_path, 'r:gz') as tar:
+        # Ensure that 'developer' exists in the tar file
+        members = [member for member in tar.getmembers() if member.name.startswith('developer')]
+        if not members:
+            raise ValueError("No 'developer' subdirectory found in the .tar.gz file.")
+        
+        for member in members:
+            member.name = os.path.basename(member.name) # Strip the 'developer' prefix
+            tar.extract(member, path=download_path)
+    
+    return download_path
+
+# Test it by downloading this file
```

Comment:
Just to get ahead of the game... when you start writing up the test code for these methods, I'd suggest using `patch`, as we do [here](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_get_gtdb.py#L93C8-L93C8), for testing your parsers etc... This way you do not have to waste time performing an actual download to test your parsing.